packages/flutter/lib/src/material/ink_splash.dart

```diff
@@ -57,6 +57,7 @@ class _InkSplashFactory extends InteractiveInkFeatureFactory {
     ShapeBorder customBorder,
     double radius,
     VoidCallback onRemoved,
+    TextDirection textDirection,
```

Comment:
This class can fail if textDirection is null - why isn't this parameter @required and asserted to be non-null?

Reply:
Added assert to constructor and required to create/constructor